linux/xdg/src/main/java/com/github/weisj/darkmode/platform/linux/xdg/FreedesktopConnection.kt

```diff
@@ -0,0 +1,81 @@
+/*
+ * MIT License
+ *
+ * Copyright (c) 2020-2022 Jannis Weis
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of this software and
+ * associated documentation files (the "Software"), to deal in the Software without restriction,
+ * including without limitation the rights to use, copy, modify, merge, publish, distribute,
+ * sublicense, and/or sell copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all copies or
+ * substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT
+ * NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+ * DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+package com.github.weisj.darkmode.platform.linux.xdg
+
+import org.freedesktop.dbus.connections.impl.DBusConnection
+import org.freedesktop.dbus.connections.impl.DBusConnectionBuilder
+import org.freedesktop.dbus.exceptions.DBusException
+import org.freedesktop.dbus.interfaces.DBusSigHandler
+import org.freedesktop.dbus.types.UInt32
+import org.freedesktop.dbus.types.Variant
+
+
+class FreedesktopConnection {
+    private val connection: DBusConnection? = try {
+        // Temporarily replace the current tread's contextClassLoader to work around dbus-java's naive service loading
+        val ccl = Thread.currentThread().contextClassLoader
+        Thread.currentThread().contextClassLoader = javaClass.classLoader
+        val conn = DBusConnectionBuilder.forSessionBus().build()
+        Thread.currentThread().contextClassLoader = ccl
+        conn
+    } catch (_: DBusException) {
+        null
+    }
```

Comment:
There already is a similar "hack" used in [Servi](https://github.com/weisJ/auto-dark-mode/blob/master/base/src/main/java/com/github/weisj/darkmode/platform/ServiceUtil.java). This should probably be extracted into a separate helper method.

Reply:
Should that go into a new class, or is there a utility class where I could add that?

Reply:
There currently isn't a good place to put it. Feel free to create a separate utility class for it.